### ISSUE TITLE
Identical AST for imports with and without a clause

### ISSUE THREAD

**Issue by azz** [eslint/typescript-eslint-parser#329]
**What version of TypeScript are you using?**

npm `2.4.0`

**What version of `typescript-eslint-parser` are you using?**

`ts-2.4` branch.

**What code were you trying to parse?**

```ts
// A
import { } from '@types/googlemaps';
// B
import '@types/googlemaps';
```

**What did you expect to happen?**

Expect `tsep(A) != tsep(B)`. The TS AST has an `importClause` field on the `ImportDeclaration`, but the tsep-produced AST has no indicator.

This is complicated though, [flow](https://astexplorer.net/#/gist/2f17206c07ba473ceaf899a4f297cc8e/2d4c7d7f1afca0f26df7560fa59917c7ff186c1b) produces identical ASTs for both A and B.

This was originally raised in https://github.com/prettier/prettier/issues/2145, where it reportedly changes the way the TypeScript checker handles the import (though I'm not sure why?)

Possibility 1. Set `specifiers` to `undefined` for case B. (ugly, ,violates [estree](https://github.com/estree/estree/blob/master/es2015.md#importdeclaration))

Possibility 2. Set `importClause` to `true` on A, `false` on B.

Possibility 3. Do nothing, this issue raised isn't valid? (I don't actually have a full understanding of the difference between A and B).

**Comment by soda0289** [comment]
I think we should just set it to null or undefined

**Comment by JamesHenry** [comment]
Yeah this is a tricky one based on the current behaviour of the various parsers. I have rasied it in espree to look for feedback.

I think it is most likely that here we will emulate the babylon way of handling it:

The first example would get `importKind: 'value'` on it, the second would not.

**Comment by JamesHenry** [comment]
@azz I think @mysticatea draws an excellent comparison between this and parenthesized expressions: https://github.com/eslint/espree/issues/357#issuecomment-338625131

I couldn't have said with 100% certainty that these are semantically equivalent off the top of my head, but it seems to be the case.

**Comment by azz** [comment]
> Possibility 3. Do nothing, this issue raised isn't valid? (I don't actually have a full understanding of the difference between A and B).

**Comment by JamesHenry** [comment]
Yeah, I guess that will be the preferred option for now. Please feel free to reopen if you have any more information about semantic differences. This might be something we can look at again in the future.
